### DISCUSSION TITLE
Displaying 404 page - prevent [slug].vue from making client side API calls for non-existent slugs

### DISCUSSION THREAD

**Discussion by lukemorell** [nuxt/nuxt#20415] [Questions]
Hi there! We are fairly new to Nuxt v3.

We have a dynamic slug file to create pages: `/pages/[slug].vue` with the following code:

```
<template>
  <main>
    <Page v-if="page" :page="page" />
  </main>
</template>

<script setup>
import Page from "@/pages/[slug]/index.vue";

const runTimeConfig = useRuntimeConfig();
const { locale } = useI18n();

const route = useRoute();

const slug = route.params.slug;

const {
  data: page,
  error,
  pending,
} = await useFetch("/pages", {
  baseURL: runTimeConfig.public.API_URL,
  method: "GET",
  query: {
    slug: slug,
    lang: locale.value || "en",
  },
  headers: {
    Authorization: `Basic ${runTimeConfig.API_TOKEN}`,
  },
  transform: (data) => data[0],
});

// Add fallback option
const fallback = !!error && error.response && error.response.status === 404;

// Add validate function
const validate = ({ params }) => !!params.slug || fallback;
</script>
```

This works well if there are pages that exist in the CMS that need a frontend page to be built (e.g. T&Cs page). When we statically generate the site, the expected html files and folders are generated, along with a 404 page that we use as a fallback.
(This 404 is served by apache using ErrorDocument 404).

However, in Nuxt v3 it seems that API calls are still being attempted even when in static mode. e.g. `example.com/asfvnsdxflgs` should fail to load any html files and display the 404 page (This work perfectly for us in Nuxt v2). Instead, it is trying to make a dynamic call to the CMS for a slug that doesn't exist and the API returns an error as expected. How do we prevent this API call from happening? Are there any other best practices in Nuxt v3 we should look out for?

Thanks!

**Comment by dulacp** [comment]
Have you tried to use the [route validation](https://nuxt.com/docs/getting-started/routing#route-validation) feature using `definePageMeta`?
